tests/testthat/test_sanity_checks.R

```diff
@@ -0,0 +1,566 @@
+context("Check sanity messages")
+
+a <<- function(x) x
+b <<- function(x) x
+f <<- function(x, y) x + y
+g <<- function(x, y = a(1)) x + y
```

Comment:
Only need single arrow assignments. `a <- function....`

Reply:
Do I? Because changing from `<<-` to `<-` causes the tests to fail. After probing, the cause of the failure is testthat's inability to find the function definitions (and the formals and default values associated with them) when running the tests with `<-`. I'm not sure how testthat scoping works (which is a terrible scenario for using `<<-` !), but `<<-` does fix things.

Reply:
I assume the fix is taking advantage of `<<-`'s behavior to assign things in the global environment environment if they are not found on the search path. This doesn't seem to overwrite what is loaded in my global environment, so I abduce that testthat runs tests in a new clean R session.

Reply:
It still seems very sketchy and we should be able to get around this.  I'm ok leaving it for now. We can add a separate issue for it.  (Reported in https://github.com/rstudio-education/gradethis/issues/104)